plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterActiveChips.php

```diff
@@ -0,0 +1,182 @@
+			$item,
+			array(
+				'title'      => '',
+				'attributes' => array(),
+			)
+		);
+
+		if ( ! $title || empty( $attributes ) ) {
+			return;
+		}
+
+		$remove_label = sprintf( 'Remove %s filter', wp_strip_all_tags( $title ) );
+		?>
+		<li class="wc-block-product-filter-active-chips__item">
+			<span class="wc-block-product-filter-active-chips__label">
+				<?php printf( '%s: %s', esc_html( $type ), wp_kses_post( $title ) ); ?>
+			</span>
+			<?php // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
+			<button class="wc-block-product-filter-active-chips__remove" aria-label="<?php echo esc_attr( $remove_label ); ?>" <?php echo $this->get_html_attributes( $attributes ); ?>>
+				<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="25" height="25" class="wc-block-product-filter-active-chips__remove-icon" aria-hidden="true" focusable="false"><path d="M12 13.06l3.712 3.713 1.061-1.06L13.061 12l3.712-3.712-1.06-1.06L12 10.938 8.288 7.227l-1.061 1.06L10.939 12l-3.712 3.712 1.06 1.061L12 13.061z"></path></svg>
+				<span class="screen-reader-text"><?php echo esc_attr( $remove_label ); ?></span>
+			</button>
+		</li>
+		<?php
+	}
+
+	/**
+	 * Build HTML attributes string from assoc array.
+	 *
+	 * @param array $attributes Attributes data as an assoc array.
+	 * @return string Escaped HTML attributes string.
+	 */
+	private function get_html_attributes( $attributes ) {
+		return array_reduce(
+			array_keys( $attributes ),
+			function ( $acc, $key ) use ( $attributes ) {
+				$acc .= sprintf( ' %1$s="%2$s"', esc_attr( $key ), esc_attr( $attributes[ $key ] ) );
+				return $acc;
+			},
+			''
+		);
+	}
+
+	/**
+	 * Parse the filter parameters from the URL.
+	 * For now we only get the global query params from the URL. In the future,
+	 * we should get the query params based on $query_id.
+	 *
+	 * @param int $query_id Query ID.
+	 * @return array Parsed filter params.
+	 */
+	private function get_filter_query_params( $query_id ) {
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$request_uri = isset( $_SERVER['REQUEST_URI'] ) ? wp_unslash( $_SERVER['REQUEST_URI'] ) : '';
+
+		$parsed_url = wp_parse_url( esc_url_raw( $request_uri ) );
+
+		if ( empty( $parsed_url['query'] ) ) {
+			return array();
+		}
+
+		parse_str( $parsed_url['query'], $url_query_params );
+
+		/**
+		 * Filters the active filter data provided by filter blocks.
+		 *
+		 * @since 11.7.0
+		 *
+		 * @param array $filter_param_keys The active filters data
+		 * @param array $url_param_keys    The query param parsed from the URL.
+		 *
+		 * @return array Active filters params.
+		 */
+		$filter_param_keys = array_unique( apply_filters( 'collection_filter_query_param_keys', array(), array_keys( $url_query_params ) ) );
+
+		return array_filter(
+			$url_query_params,
+			function ( $key ) use ( $filter_param_keys ) {
+				return in_array( $key, $filter_param_keys, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+	}
```

Comment:
So since we decided to create a separate chips block just for active chips, I didn't think it would be reused since it is much more specific to Active filters so I just thought to have the logic live there.

Reply:
If we do this, we break the convention that the display style block is swappable as this block is not simple enough. If we pass the processed list down from the Active block, then Active block can have other display styles (by us or by 3rd parties).
One possible argument here is: that it sounds premature, but we're already doing the same for other filter blocks.